plugins/notifications-backend/src/service/handlers.ts

```diff
@@ -89,30 +136,39 @@ export async function getNotifications(
     );
   }
 
-  const query = dbClient('messages').select('*');
+  if (filter.messageType !== 'system' && !filter.user) {
+    throw new Error(
+      'user parameter is missing in request',
+    );
+  }
 
-  query.orderBy('created', 'asc');
+  let userGroups: string[] = [];
+  
+  if (filter.user) {
+    userGroups = await getUserGroups(catalogClient, filter.user)
+  }
 
-  addFilter(query, filter);
+  const query = createQuery(dbClient, filter, userGroups);
 
   if (pageNumber > 0) {
     query.limit(pageSize).offset((pageNumber - 1) * pageSize);
   }
 
-  const notifications = await query.then(messages =>
+  const notifications = await query.select('*').then(messages =>
     messages.map(message => {
       const notification: Notification = {
         id: message.id,
         created: message.created,
-        readByUser: false,
+        isSystem: message.is_system,
+        readByUser: message.read ? message.read : false,
```

Comment:
nit: 
```suggestion
        readByUser: !!message.read,
```

Reply:
i think it is confusing

Reply:
It's a nit pick, up to you.
The suggested syntax is much more common to JS devs than the first one in similar cases.